mempool/cat/store.go

```diff
@@ -0,0 +1,158 @@
+package cat
+
+import (
+	"sync"
+	"time"
+
+	"github.com/tendermint/tendermint/types"
+)
+
+// simple, thread-safe in memory store for transactions
+type store struct {
+	mtx   sync.RWMutex
+	bytes int64
+	txs   map[types.TxKey]*wrappedTx
+}
+
+func newStore() *store {
+	return &store{
+		bytes: 0,
+		txs:   make(map[types.TxKey]*wrappedTx),
+	}
+}
+
+func (s *store) set(wtx *wrappedTx) bool {
+	if wtx == nil {
```

Comment:
Yeah that's a good point. I think it's just a defensive thing but you're right that if it did happen for some reason it would go undetected. Perhaps it's better if we panic.